### PR TITLE
added cgroup updates and oomKillDisable annotation

### PR DESCRIPTION
**Pull request by pdgetrf**
(1) added updating pod level cgroups aggregration based on pod resource spec
(2) added oomKillDisable annotation, passed down to pod level cgroup file memory.oom_control

Tested on a local environment

### CODE REVIEW

File: pkg/kubelet/cm/pod_container_manager_linux.go [hunk 1]
```diff
@@ -75,21 +75,35 @@ func (m *podContainerManagerImpl) Exists(pod *v1.Pod) bool {
...
+	containerConfig := &CgroupConfig{
+		Name:               podContainerName,
+		ResourceParameters: ResourceConfigForPod(pod, m.enforceCPULimits),
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.SupportPodPidsLimit) && m.podPidsLimit > 0 {
+		containerConfig.ResourceParameters.PodPidsLimit = &m.podPidsLimit
+	}
+
+	// parse oomKillDisable flag
+	oomKillDisable := pod.ObjectMeta.Annotations["oomKillDisable"]
```
Comment (vinaykul) [inline pkg/kubelet/cm/pod_container_manager_linux.go]: lets define this as a const in some location

File: pkg/kubelet/cm/types.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ type ResourceConfig struct {
 	HugePageLimit map[int64]int64
 	// Maximum number of pids
 	PodPidsLimit *int64
+	// OOM Kill Configuration
```
Comment (vinaykul) [inline pkg/kubelet/cm/types.go]: Lets clarify the comment as something like "Configuration of out-of-memory process kill setting for pod"

**Review by vinaykul — CHANGES_REQUESTED** [review]
Couple of minor changes

File: pkg/kubelet/cm/types.go (hunk 2 above)
Comment (pdgetrf) [reply pkg/kubelet/cm/types.go]: fixed

File: pkg/kubelet/cm/pod_container_manager_linux.go (hunk 1 above)
Comment (pdgetrf) [reply pkg/kubelet/cm/pod_container_manager_linux.go]: fixed

**Review by vinaykul — APPROVED** [review]

**Pull request merged**